frontends/telegram/app/handlers/admin/subscriptions.py

```diff
@@ -0,0 +1,61 @@
+import requests
+
+from aiogram import Router, F
+from aiogram.filters import StateFilter
+from aiogram.fsm.context import FSMContext
+from aiogram.types import Message
+
+from frontends.telegram.app.states import PublishSubscription
+from frontends.telegram.app.settings import Settings
+from frontends.telegram.app.keyboards.reply.admin import main_kb
+from frontends.telegram.app.filters.role import RoleFilter
+from frontends.telegram.app.models.role import UserRole
+
+router = Router(name=__name__)
+# Apply RoleFilter to all router's handlers
+router.message.filter(RoleFilter(UserRole.ADMIN))
+router.callback_query.filter(RoleFilter(UserRole.ADMIN))
```

Comment:
Same may be here

Reply:
Here we apply filter for events, no middlewares. And filters applies for messages and callbacks separately just like middlewares.
We can also apply filters to each handler using a decorator, but it is much easier to apply filters to the entire router.